File: src/api/MoRS.ManagementSystem.Application/DTOs/Payment/CreatePaymentRequest.cs
using MoRS.ManagementSystem.Domain.Enums;

namespace MoRS.ManagementSystem.Application.DTOs.Payment;

public class CreatePaymentRequest
{
    public decimal Amount { get; set; }
    public PaymentStatus Status { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public required string TransactionId { get; set; }
    public int UserId { get; set; }
}

File: src/mobile/lib/models/payment/create_payment_request.dart
class CreatePaymentRequest {
  final double amount;
  final int status; 
  final int
  paymentMethod;
  final String transactionId;
  final int userId;

  CreatePaymentRequest({
    required this.amount,
    required this.status,
    required this.paymentMethod,
    required this.transactionId,
    required this.userId,
  });

  Map<String, dynamic> toJson() {
    return {
      'Amount': amount,
      'Status': status,
      'PaymentMethod': paymentMethod,
      'TransactionId': transactionId,
      'UserId': userId,
    };
  }
}
